htdocs/comm/propal/class/propal.class.php

```diff
@@ -441,7 +458,7 @@ function addline($desc, $pu_ht, $qty, $txtva, $txlocaltax1=0.0, $txlocaltax2=0.0
             $localtaxes_type=getLocalTaxesFromRate($txtva,0,$this->thirdparty,$mysoc);
             $txtva = preg_replace('/\s*\(.*\)/','',$txtva);  // Remove code into vatrate.
             
-            $tabprice=calcul_price_total($qty, $pu, $remise_percent, $txtva, $txlocaltax1, $txlocaltax2, 0, $price_base_type, $info_bits, $product_type, $mysoc, $localtaxes_type);
+            $tabprice=calcul_price_total($qty * $durationqty, $pu, $remise_percent, $txtva, $txlocaltax1, $txlocaltax2, 0, $price_base_type, $info_bits, $product_type, $mysoc, $localtaxes_type);
```

Comment:
This means you decide to work with quantity that are decimals and calculated with a prorata of time defined by start and end date.
All users won't want to work this way, so this must be an option.
Also this will bock the PR integration, for all the 19 modified files. it is recommanded when a lot of files are impacted, to split changes into smaller PR than possible, for example a PR just to add the control that date start is before end date is enough to be one standalone PR.